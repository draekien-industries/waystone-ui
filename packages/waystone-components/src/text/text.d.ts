/** @jsxImportSource theme-ui */
import React, { FC } from 'react';
import { BoxProps } from 'theme-ui';
import { TextVariant } from '@waystone/core';
import { HasColor } from '../common/interfaces';
/**
 * Props for rendering the {@link Text} component.
 * @extends React.HTMLAttributes<HTMLDivElement>
 * @extends BoxProps
 */
export interface TextProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'color'>, Omit<BoxProps, 'color'>, HasColor {
    /**
     * Should the text component place itself on the same line
     * as the preceeding component (if possible).
     * @default false
     */
    inline?: boolean;
    /**
     * The text variant to render.
     * @override
     * @default 'body'
     */
    variant?: TextVariant;
}
/**
 * Renders a text component.
 * @returns The text component
 */
export declare const Text: FC<TextProps>;
//# sourceMappingURL=text.d.ts.map