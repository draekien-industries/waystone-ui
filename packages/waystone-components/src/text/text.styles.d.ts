import { ThemeUIStyleObject } from 'theme-ui';
import { HasColor, HasVariant } from '../common/interfaces';
/** @internal */
export interface TextCssOptions extends HasVariant, HasColor {
    inline: boolean;
}
/** @internal */
export declare const textCss: (options: TextCssOptions) => ThemeUIStyleObject;
//# sourceMappingURL=text.styles.d.ts.map