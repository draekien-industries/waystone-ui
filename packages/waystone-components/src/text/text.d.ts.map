{"version":3,"file":"text.d.ts","sourceRoot":"","sources":["text.tsx"],"names":[],"mappings":"AAAA,gCAAgC;AAChC,OAAO,KAAK,EAAE,EAAE,EAAE,EAAE,MAAM,OAAO,CAAC;AAClC,OAAO,EAAuB,QAAQ,EAAE,MAAM,UAAU,CAAC;AACzD,OAAO,EAAE,WAAW,EAAE,MAAM,gBAAgB,CAAC;AAE7C,OAAO,EAAE,QAAQ,EAAE,MAAM,sBAAsB,CAAC;AAEhD;;;;GAIG;AACH,MAAM,WAAW,SACf,SAAQ,IAAI,CAAC,KAAK,CAAC,cAAc,CAAC,cAAc,CAAC,EAAE,OAAO,CAAC,EACzD,IAAI,CAAC,QAAQ,EAAE,OAAO,CAAC,EACvB,QAAQ;IACV;;;;OAIG;IACH,MAAM,CAAC,EAAE,OAAO,CAAC;IACjB;;;;OAIG;IACH,OAAO,CAAC,EAAE,WAAW,CAAC;CACvB;AAED;;;GAGG;AACH,eAAO,MAAM,IAAI,EAAE,EAAE,CAAC,SAAS,CAU9B,CAAC"}