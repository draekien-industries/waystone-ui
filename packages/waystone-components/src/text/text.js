var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { jsx as _jsx } from "theme-ui/jsx-runtime";
import { Text as ThemeUiText } from 'theme-ui';
import { textCss } from './text.styles';
/**
 * Renders a text component.
 * @returns The text component
 */
export const Text = (_a) => {
    var { children, inline = false, variant = 'body', color } = _a, rest = __rest(_a, ["children", "inline", "variant", "color"]);
    return (_jsx(ThemeUiText, Object.assign({ sx: textCss({ inline, variant, color }) }, rest, { children: children })));
};
