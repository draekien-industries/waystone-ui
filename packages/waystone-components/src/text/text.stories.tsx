import React from 'react';
import { Meta, Story } from '@storybook/react';
import { TextVariant } from '@waystone/core';
import { Text, TextProps } from './text';

export default {
  title: 'Components/Text',
  component: Text,
  args: {
    variant: 'body',
    inline: false,
    children:
      'Lorem, ipsum dolor sit amet consectetur adipisicing elit. ' +
      'Non dolorum facilis similique commodi, officiis error quis molestiae in repellat, ' +
      'minus delectus nesciunt, voluptas quos autem? Nostrum quidem mollitia nemo quas!',
  },
} as Meta;

const Template: Story<TextProps> = (args) => <Text {...args} />;

export const Default = Template.bind({});

const VariantTemplate: Story<TextProps> = (args) => {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const { variant, children, ...rest } = args;

  return (
    <div>
      {Object.keys(TextVariant).map((textVariant, index) => (
        <Text key={index} variant={textVariant as TextVariant} {...rest}>
          {variant}
        </Text>
      ))}
    </div>
  );
};

export const WithVariant = VariantTemplate.bind({});

WithVariant.parameters = {
  docs: {
    description: {
      story:
        'These are the text variants that have been pre-defined in the theme.',
    },
  },
};

export const WithColor = Template.bind({});

WithColor.parameters = {
  docs: {
    description: {
      story:
        'The `colors` prop determines the color of the text that is rendered. ' +
        'You can use any of the colors included in the theme ' +
        'or your own color.',
    },
  },
};

WithColor.args = {
  color: 's-400',
};

const InlineTemplate: Story<TextProps> = (args) => {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const { color, ...rest } = args;

  return (
    <div>
      <Text {...rest} color="s-400" />
      <Text {...rest} color="h-400" />
    </div>
  );
};

export const WithInline = InlineTemplate.bind({});

WithInline.parameters = {
  docs: {
    description: {
      story:
        'When the `inline` prop is set to `true`, ' +
        'the component will render with `inline-block` as the CSS display value.',
    },
  },
};

WithInline.args = {
  inline: true,
  children: 'Lorem ipsum.',
};
