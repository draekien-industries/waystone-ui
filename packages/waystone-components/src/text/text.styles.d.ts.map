{"version":3,"file":"text.styles.d.ts","sourceRoot":"","sources":["text.styles.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,kBAAkB,EAAE,MAAM,UAAU,CAAC;AAC9C,OAAO,EAAE,QAAQ,EAAE,UAAU,EAAE,MAAM,sBAAsB,CAAC;AAE5D,gBAAgB;AAChB,MAAM,WAAW,cAAe,SAAQ,UAAU,EAAE,QAAQ;IAC1D,MAAM,EAAE,OAAO,CAAC;CACjB;AAED,gBAAgB;AAChB,eAAO,MAAM,OAAO,EAAE,CAAC,OAAO,EAAE,cAAc,KAAK,kBAgBlD,CAAC"}