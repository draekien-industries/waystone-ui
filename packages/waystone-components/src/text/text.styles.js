/** @internal */
export const textCss = ({ inline, variant, color, }) => {
    const css = {
        display: 'block',
        variant: `text.${variant}`,
        color: color || 'inherit',
    };
    if (inline) {
        css.display = 'inline-block';
    }
    return css;
};
