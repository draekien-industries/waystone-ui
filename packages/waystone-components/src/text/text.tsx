/** @jsxImportSource theme-ui */
import React from 'react';
import { Text as ThemeUiText } from 'theme-ui';
import { TextVariant } from '@waystone/core';
import { textCss } from './text.styles';
import { HasColor, HasVariant } from '../common/interfaces';

/**
 * Props for rendering the {@link Text} component.
 * @extends HasColor
 * @extends HasVariant
 */
export interface TextProps extends HasColor, HasVariant {
  /**
   * The type of element to render onto the DOM.
   * @default 'span'
   */
  as?: React.ElementType;
  /**
   * Should the text component place itself on the same line
   * as the preceeding component (if possible).
   * @default false
   */
  inline?: boolean;
  /**
   * The text variant to render.
   * @override
   * @default 'body'
   */
  variant?: TextVariant;
  /**
   * The components to render as children.
   */
  children: React.ReactNode;
}

/**
 * A component for rendering text onto the DOM with props for setting the text variant, color,
 * HTML element type.
 */
export const Text = ({
  children,
  as = 'span',
  inline = false,
  variant = 'body',
  color,
  ...rest
}: TextProps) => (
  <ThemeUiText sx={textCss({ inline, variant, color })} as={as} {...rest}>
    {children}
  </ThemeUiText>
);
