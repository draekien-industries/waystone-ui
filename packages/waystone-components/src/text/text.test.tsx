import { render } from '@waystone/testing-library';
import { Text } from './text';

describe('The Text component', () => {
  it('Matches snapshot', () => {
    const { container } = render(<Text>text</Text>);

    expect(container).toMatchSnapshot();
  });
});
