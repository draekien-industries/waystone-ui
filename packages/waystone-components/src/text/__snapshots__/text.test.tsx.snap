// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Text component Matches snapshot 1`] = `
<div>
  <span
    class="css-1k3tifs-Text"
  >
    text
  </span>
</div>
`;
